### ISSUE TITLE
Question about evaluation results on OPV2V

### ISSUE THREAD

**Issue by Bosszhe** [starfdu1418/SCOPE#4]
Thanks for your contribution to the community, I am really interested in your work.  But I met some problems when I tried to reproduce the results in the paper, Hope you can give me some advice.

I just downloaded the checkpoint from the Google Drive [[OPV2V]](https://drive.google.com/drive/folders/1ezHtKYKD8a1os0HhPkRKkdeoxCjpeNf_?usp=sharing) and test the model with:

`OPV2V dataset: python v2xvit/tools/inference.py --model_dir ${CHECKPOINT_FOLDER} --eval_epoch 34`

I got the evaluation results on validate split as follows:

![image](https://github.com/starfdu1418/SCOPE/assets/34713622/05f2de0f-0b3b-4f80-ac76-0508092192f2)

Then I tried to train the model from the sketch, but I only got results lower than yours: 

![image](https://github.com/starfdu1418/SCOPE/assets/34713622/5e9ae415-25dc-47fe-b145-0a5d30e718ea)

---
What's more, I want to verify whether the results of OPV2V in Table 1 in the paper are inferred from the 'test' split.
![image](https://github.com/starfdu1418/SCOPE/assets/34713622/1a59cc1a-59c5-431f-a238-7e32dfd84738)

---

Here is the config.yaml
```
data_augment:
- ALONG_AXIS_LIST:
  - x
  NAME: random_world_flip
- NAME: random_world_rotation
  WORLD_ROT_ANGLE:
  - -0.78539816
  - 0.78539816
- NAME: random_world_scaling
  WORLD_SCALE_RANGE:
  - 0.95
  - 1.05
fusion:
  args:
    cur_ego_pose_flag: true
  core_method: IntermediateFusionDataset
loss:
  args:
    cls_weight: 1.0
    reg: 2.0
  core_method: point_pillar_loss
lr_scheduler:
  core_method: multistep
  gamma: 0.1
  step_size:
  - 10
  - 20
model:
  args:
    anchor_number: 2
    backbone_fix: false
    base_bev_backbone:
      compression: 0
      layer_nums: &id001
      - 3
      - 4
      - 5
      layer_strides:
      - 2
      - 2
      - 2
      num_filters: &id002
      - 64
      - 128
      - 256
      num_upsample_filter:
      - 128
      - 128
      - 128
      resnet: true
      upsample_strides:
      - 1
      - 2
      - 4
      voxel_size: &id003
      - 0.4
      - 0.4
      - 4
    compression: 0
    fusion_args:
      agg_operator:
        depth: 1
        embed_pos: None
        feature_dim: 256
        hetero: false
        mode: Deform
        n_head: 8
        with_scm: false
        with_spe: false
      attn_communication:
        channel: 64
        compression_rate: 16
        gaussian_smooth:
          c_sigma: 1.0
          k_size: 5
        mode: topk
        percent: 1.0
        thre: 0.01
      communication:
        gaussian_smooth:
          c_sigma: 1.0
          k_size: 5
        thre: 0.01
      downsample_rate: 1
      dropout_rate: 0
      frame: 1
      in_channels: 256
      layer_nums: *id001
      multi_scale: true
      n_head: 8
      num_filters: *id002
      only_attention: true
      only_ego: false
      temporal: self
      temporal_fusion:
        channel: 256
        gate: false
        height: 100
        layers: 1
        n_head: 5
        rte_ratio: 1
        width: 352
      voxel_size: *id003
    lidar_range: &id004
    - -140.8
    - -40
    - -3
    - 140.8
    - 40
    - 1
    max_cav: 5
    pillar_vfe:
      num_filters:
      - 64
      use_absolute_xyz: true
      use_norm: true
      with_distance: false
    point_pillar_scatter:
      grid_size: !!python/object/apply:numpy.core.multiarray._reconstruct
        args:
        - !!python/name:numpy.ndarray ''
        - !!python/tuple
          - 0
        - !!binary |
          Yg==
        state: !!python/tuple
        - 1
        - !!python/tuple
          - 3
        - !!python/object/apply:numpy.dtype
          args:
          - i8
          - false
          - true
          state: !!python/tuple
          - 3
          - <
          - null
          - null
          - null
          - -1
          - -1
          - 0
        - false
        - !!binary |
          wAIAAAAAAADIAAAAAAAAAAEAAAAAAAAA
      num_features: 64
    shrink_header:
      dim:
      - 256
      input_dim: 384
      kernal_size:
      - 3
      padding:
      - 1
      stride:
      - 1
    voxel_size: *id003
  core_method: point_pillar_scope
name: opv2v_deformable_transformer_multiscale
optimizer:
  args:
    eps: 1.0e-10
    weight_decay: 0.0001
  core_method: Adam
  lr: 0.002
postprocess:
  anchor_args:
    D: 1
    H: 200
    W: 704
    cav_lidar_range: *id004
    feature_stride: 2
    h: 1.56
    l: 3.9
    num: 2
    r:
    - 0
    - 90
    vd: 4
    vh: 0.4
    vw: 0.4
    w: 1.6
  core_method: VoxelPostprocessor
  gt_range: *id004
  max_num: 100
  nms_thresh: 0.15
  order: hwl
  target_args:
    neg_threshold: 0.45
    pos_threshold: 0.6
    score_threshold: 0.23
preprocess:
  args:
    max_points_per_voxel: 32
    max_voxel_test: 70000
    max_voxel_train: 32000
    voxel_size: *id003
  cav_lidar_range: *id004
  core_method: SpVoxelPreprocessor
root_dir: ./data/OPV2V/train
train_params:
  batch_size: 1
  epoches: 22
  eval_freq: 1
  frame: 1
  max_cav: 5
  save_freq: 1
validate_dir: ./data/OPV2V/validate
wild_setting:
  async: false
  async_overhead: 100
  backbone_delay: 10
  data_size: 1.06
  loc_err: false
  ryp_std: 0.2
  seed: 20
  transmission_speed: 27
  xyz_std: 0.2
yaml_parser: load_point_pillar_params

```

**Comment by muse1418** [comment]
1. Obviously, you used validation split. We inferred pre-trained models with test split as stated in README.md and our paper.
2. The accuracy of our pre-trained model is shown here:
![result](https://github.com/starfdu1418/SCOPE/assets/74818266/4246eb27-b5c2-4847-9748-2772587c2562)
